queue/abstract/driver/utubettl.lua

```diff
@@ -87,18 +102,83 @@ function tube.create_space(space_name, opts)
         type = 'tree',
         parts = {i_status, str_type(), i_utube, str_type(), i_id, num_type()}
     })
+    space:create_index('utube_pri', {
+        type = 'tree',
+        parts = {i_status, str_type(), i_utube, str_type(), i_pri, num_type(), i_id, num_type()}
+    })
     return space
 end
 
 local delayed_state = { state.DELAYED }
 local ttl_states    = { state.READY, state.BURIED }
 local ttr_state     = { state.TAKEN }
 
+local function put_next_ready(self, utube)
+    local taken = self.space.index.utube:min{state.TAKEN, utube}
+    if taken == nil or taken[i_status] ~= state.TAKEN then
+        local next_task = self.space.index.utube_pri:min{state.READY, utube}
+        if next_task == nil or next_task[i_status] ~= state.READY then
+            return
+        end
+        -- Ignoring ER_TUPLE_FOUND error, if a tuple with the same task_id
+        -- or utube name is already in the space.
+        -- Note that both task_id and utube indexes are unique, so there will be
+        -- no duplicates: each task_id can occur in the space not more than once,
+        -- there can be no more than one task from each utube in a space.
+        pcall(self.space_ready_buffer.insert, self.space_ready_buffer, {next_task[i_id], utube, next_task[i_pri]})
+    end
+end
+
+local function put_ready(self, id, utube, pri)
+    local taken = self.space.index.utube:min{state.TAKEN, utube}
+    if taken == nil or taken[i_status] ~= state.TAKEN then
+        local current_task = self.space.index.utube_pri:min{state.READY, utube}
+        if current_task == nil or current_task[i_status] ~= state.READY or current_task[i_pri] < pri then
+            return
+        end
+        if current_task ~= nil and current_task[i_pri] > pri then
```

Comment:
`>` or `<`? Please, ensure that priority works as before.
```suggestion
        if current_task ~= nil and current_task[i_pri] > pri then
```

Reply:
Priority works the same as nice number -- less numbers have more priority. So if in the ready space currently exists a task with higher number (and thus, lower priority), we need to delete it before putting new task (with higher priority).